### PR TITLE
Add drag and drop avatar uploading

### LINKED ISSUE #107: Add drag&drop for uploading new avatar
That would be easier if you add such option.

### PR DESCRIPTION
**Pull request by josephnle**
- Turned the avatar into a dropzone which takes uploads. Used native APIs.
- Refactor uploading a bit to allow both the file picker and drag & drop method to use the same upload code
- Used the same styling as loading/open/hover when user drags their avatar into the dropzone.

Closes #107

### CODE REVIEW

**Comment by tobscure** [comment]
😍

File: js/forum/src/components/AvatarEditor.js [hunk 1]
```diff
@@ -78,6 +90,40 @@ export default class AvatarEditor extends Component {
...
 
   /**
+   * Enable dragover style
+   *
+   * @param {Event} e
+   */
+  enableDragover(e) {
+    e.preventDefault();
+    e.stopPropagation();
+    this.isDraggedOver = true;
```
Comment (franzliedke) [inline js/forum/src/components/AvatarEditor.js]: Do we need redraws here?

**Comment by franzliedke** [comment]
Wow, awesome, great job!

File: js/forum/src/components/AvatarEditor.js (hunk 1 above)
Comment (tobscure) [reply js/forum/src/components/AvatarEditor.js]: https://mithril.js.org/autoredraw.html

**Comment by tobscure** [comment]
Thanks for your awesome work @josephnle!

**Pull request merged**